apps/staking/hooks/useNetworkBalances.tsx

```diff
@@ -68,20 +82,45 @@ export const useNetworkBalances = (params?: { addressOverride?: Address }) => {
       lockedStakes,
       claimableRewards,
       claimableStakes,
-      networkClaimRemainingInCurrentPeriod,
-      networkClaimPeriodEnd: networkClaimCurrentPeriodEnd,
     };
   }, [data]);
 
   const canClaim =
     status === 'success' && parsedData.unclaimed >= BigInt(HANDRAIL_THRESHOLD.CLAIM_REWARDS_AMOUNT);
 
-  const isClaimOverLimit = parsedData.unclaimed > parsedData.networkClaimRemainingInCurrentPeriod;
+  const claimPeriodData = useMemo(() => {
+    let isClaimOverLimit = false;
+    let networkClaimPeriodEnd: number | null = null;
+
+    if (claimCycleStatus === 'success' && canClaim) {
+      const networkCycleTotalAfterClaim = currentClaimTotal + parsedData.unclaimed;
+      if (networkCycleTotalAfterClaim > claimThreshold || forceClaimOverThreshold) {
+        isClaimOverLimit = true;
+        const now = Math.trunc(Date.now() / 1000);
+        const cc = bigIntToNumber(claimCycle, 0);
```

Comment:
Should we guard against claimCycle being 0 here?

Reply:
good spot!